internal/core/component.rs

```diff
@@ -32,17 +43,35 @@ pub struct ComponentVTable {
         index: usize,
     ) -> core::pin::Pin<VRef<ItemVTable>>,
 
+    /// Return the range of indices below the dynamic `ItemTreeNode` at `index`
+    pub get_subtree_range:
+        extern "C" fn(core::pin::Pin<VRef<ComponentVTable>>, index: usize) -> IndexRange,
+
+    /// Return the `ComponentRc` at `subindex` below the dynamic `ItemTreeNode` at `index`
+    pub get_subtree_component: extern "C" fn(
+        core::pin::Pin<VRef<ComponentVTable>>,
+        index: usize,
+        subindex: usize,
+        result: &mut vtable::VWeak<ComponentVTable, Dyn>,
+    ),
+
     /// Return the item tree that is defined by this `Component`.
     /// The return value is an item weak because it can be null if there is no parent.
     /// And the return value is passed by &mut because ItemWeak has a destructor
     pub get_item_tree: extern "C" fn(core::pin::Pin<VRef<ComponentVTable>>) -> Slice<ItemTreeNode>,
 
+    // FIXME: This does return an invalid ItemWeak now that points to the parent repeater!
+    // FIXME: Get rid of the index and make this always return the "Item" that connects this component
+    //        to its parent-component?
```

Comment:
i don't really understand what this is about.

Reply:
I would like this to end up as somehting along this line:

`fn parent_subtree(ComponentRc) -> Option<(ComponentRc, usize)>,` (in FFI-friendly;-)

and then make this unconditionally return the ItemWeak we currently return when the index is `0`.

There is no need to duplicate the code to find the parent item when staying inside the components item tree in all the different generated code we have! So this is the only interesting bit of information this method provides right now.